src/util/fs_helpers.cpp

```diff
@@ -206,17 +206,28 @@ void AllocateFileRange(FILE* file, unsigned int offset, unsigned int length)
     if (0 == posix_fallocate(fileno(file), 0, nEndPos)) return;
 #endif
     // Fallback version
-    // TODO: just write one byte per block
-    static const char buf[65536] = {};
-    if (fseek(file, offset, SEEK_SET)) {
+    if (fseek(file, 0, SEEK_END)) {
+        return;
+    }
+    const int64_t filesize = std::ftell(file);
+    if (filesize < 0) {
+        return;
+    }
+    if (offset + length <= static_cast<unsigned int>(filesize)) {
         return;
     }
-    while (length > 0) {
+    unsigned int inc_size = offset + length - static_cast<unsigned int>(filesize);
```

Comment:
Change of behaviour: This can allocate before `offset` if the file is smaller. If that's desired behaviour, there's no point to the `offset` parameter at all...?